rust/gitxetcore/src/command/login.rs

```diff
@@ -82,10 +87,167 @@ fn apply_config(
                 axe_config.axe_code = Some(axe_key);
             }
         }
+        if let Some(aws_access_key) = auth_check.aws_access_key {
+            if !aws_access_key.is_empty() {
+                user.aws_access_key = Some(aws_access_key);
+            }
+        }
+        if let Some(aws_secret_key) = auth_check.aws_secret_key {
+            if !aws_secret_key.is_empty() {
+                user.aws_secret_key = Some(aws_secret_key);
+            }
+        }
     }
     Ok(())
 }
 
+// Run AWS CLI to write an access key and secret key
+// Returns the profilename we wrote to
+fn write_aws_config(cfg: &Cfg, host: &str) -> errors::Result<String> {
+    if cfg.user.is_none() {
+        return Err(errors::GitXetRepoError::Other(
+            "No user configuration".to_string(),
+        ));
+    }
+    let user = &cfg.user.as_ref().unwrap();
+    let aws_access_key = user
+        .aws_access_key
+        .as_ref()
+        .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+            "Not authenticated. Run git-xet login first"
+        )))?;
+    let aws_secret_key = user
+        .aws_secret_key
+        .as_ref()
+        .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+            "Not authenticated. Run git-xet login first"
+        )))?;
+    // dots and dashes are ok
+    // this makes the profile name slightly more readable
+    let mut profilename = host.to_string();
+    profilename.retain(|x| x.is_alphanumeric() || x == '.' || x == '-');
+    // strip the .com so xethub.com just becomes xethub
+    if profilename.ends_with(".com") {
+        profilename = profilename[..profilename.len() - 4].to_string();
+    }
+    let output = Command::new("aws")
+        .args([
+            "configure",
+            "--profile",
+            &profilename,
+            "set",
+            "aws_access_key_id",
+            &aws_access_key,
+        ])
+        .status()?;
+    if !output.success() {
+        return Err(errors::GitXetRepoError::Other(
+            "Unable to run aws cli".to_string(),
+        ));
+    }
+    let output = Command::new("aws")
+        .args([
+            "configure",
+            "--profile",
+            &profilename,
+            "set",
+            "aws_secret_access_key",
+            &aws_secret_key,
+        ])
+        .status()?;
+    if !output.success() {
+        return Err(errors::GitXetRepoError::Other(
+            "Unable to run aws cli".to_string(),
+        ));
+    }
+    Ok(profilename.to_string())
+}
+
+/// Prints the AWS Config stdout.
+/// Does not do anything if there is no user config.
+fn print_s3_config(cfg: &Cfg) -> errors::Result<()> {
+    if let Some(ref user) = cfg.user {
+        let aws_access_key =
+            &user
+                .aws_access_key
+                .as_ref()
+                .ok_or(errors::GitXetRepoError::AuthError(anyhow!(
+                    "Not authenticated. Run git-xet login first"
```

Comment:
IDK if it matters, but with all the error messages being the same, we won't know which line caused an error if a user reports it.